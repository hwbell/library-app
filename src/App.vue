<template>
  <div id="app">
    <div id="content">
      <Navigator/>

      <!--This is the master component for all things flowing from the user's search 
        Search will contain results of the the search, as well as the user's
      readingList, which is selected from the results-->
      <Search/>

      <!-- <Suggestions/> -->
    </div>
  </div>
</template>

<script>

import Vue from "vue";
// components
import Navigator from "./components/Navigator";
import Search from "./components/Search";

// tools
import BootstrapVue from "bootstrap-vue";

Vue.use(BootstrapVue);

export default {
  name: "app",
  components: {
    Navigator,
    Search,
    // Suggestions
  },
  methods: {},
  data() {
    return {
      newSearch: ""
    };
  }
};
</script>

<style lang="scss">
@import url("https://fonts.googleapis.com/css?family=Quicksand");
@import "node_modules/bootstrap/scss/bootstrap";
@import "node_modules/bootstrap-vue/src/index.scss";

$link-blue: rgb(0, 119, 255);

* {
  font-family: Quicksand;
  font-weight: lighter;
}
* > a {
  cursor: pointer;
  transition: all 0.3s ease;
  // text-decoration: none;
  color: $link-blue;
  &:hover {
    color: $link-blue;
  }
}
#app {
  // z-index: 0;
  min-height: 100vh;
  background: url("assets/books-bg.jpeg") no-repeat;
  background-size: 2000px 400px;
  background-position: 0% 120%;
  background-attachment: fixed;
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
}
#content {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: rgb(255,255,255, 0.75)
}
.fade-enter-active,
.fade-leave-active {
  transition: all 0.3s;
}
.fade-enter,
.fade-leave-to {
  opacity: 0;
}
.logo {
  width: 50px;
  height: 50px;
}
</style>
