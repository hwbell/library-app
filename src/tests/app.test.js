// Import Vue and the component being tested
import { mount } from '@vue/test-utils'

import Vue from 'vue'
import App from '../App.vue';

describe('App', () => {

  it('renders correctly', () => {
    const wrapper = mount(App)
    expect(wrapper.element).toMatchSnapshot()
  })

  const wrapper = mount(App)
  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('contains app selector', () => {
    expect(wrapper.html()).toContain('id="app"')
  })
  it('contains content selector', () => {
    expect(wrapper.html()).toContain('id="content"')
  })
  it('contains navigator selector', () => {
    expect(wrapper.html()).toContain('id="navigator"')
  })
  it('contains search selector', () => {
    expect(wrapper.html()).toContain('id="search"')
  })

})

