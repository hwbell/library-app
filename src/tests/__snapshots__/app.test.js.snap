// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  id="app"
>
  <div
    id="content"
  >
    <div
      id="navigator"
    >
      <p
        class="app-name"
      >
        Your Library
      </p>
       
      <mqshow
        if="sm"
      >
        <div
          class="dropdown btn-group b-dropdown m-md-2 dropdown"
          id="dropdown"
        >
          <!---->
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="btn dropdown-toggle btn-primary"
            id="dropdown__BV_toggle_"
            type="button"
          >
            Go to
          </button>
          <ul
            aria-labelledby="dropdown__BV_toggle_"
            class="dropdown-menu"
            role="menu"
            tabindex="-1"
          >
            <li>
              <a
                class="dropdown-item"
                href="#"
                role="menuitem"
                target="_self"
              >
                <a
                  href="#reading-list"
                >
                  my list
                </a>
              </a>
            </li>
             
            <li>
              <a
                class="dropdown-item"
                href="#"
                role="menuitem"
                target="_self"
              >
                <a
                  href="#search"
                >
                  find more
                </a>
              </a>
            </li>
          </ul>
        </div>
      </mqshow>
       
      <mqshow
        if="md!"
      >
        <div
          class="links-holder"
        >
          <p
            class="nav-link"
          >
            <a
              href="#reading-list"
            >
              my list
            </a>
          </p>
           
          <p
            class="nav-link"
          >
            <a
              href="#search"
            >
              find more
            </a>
          </p>
        </div>
      </mqshow>
    </div>
     
    <div
      id="search"
    >
      <p
        id="ask"
      >
        What do you feel like reading?
      </p>
       
      <div
        id="input-holder"
      >
        <input
          id="search-input"
          placeholder="book, author, subject, keyword"
          type="text"
        />
         
        <img
          alt=""
          id="search-icon"
          src="../assets/search.png"
        />
      </div>
       
      <!---->
       
      <div
        class="results row"
      />
       
      <p
        class="subtitle"
        id="reading-list"
      >
        Your Reading List
      </p>
       
      <p
        class="list-intro"
      >
        
    Your list is empty! Search for some books from Google, add them to your
    list, and they will appear here. 
  
      </p>
       
      <div
        class="results row"
      />
       
      <!---->
    </div>
  </div>
</div>
`;
